### PR TITLE
fix(GlobalHeader): fix mobile menu for screen sizes btwn 1024-1200px

### PR DESCRIPTION
**Pull request by saghdaey**
### Overview

- match overlay display media query to mobile header media query
- fix issue where app header dropdown wasn't always displaying (i.e. had a height of 0) by updating the layout hook dependencies to include isOpen. When isOpen is true that means animation complete + height is no longer 0

### Testing

Visit production story book:
https://gamut.codecademy.com/?path=/story/organisms-globalheader--free-user-pro-upgrade
open "canvas" view
select "MD" screen size
click on hamburger menu button
see that the header disappears

Now run my branch locally
Visit http://localhost:6006/?path=/story/organisms-globalheader--free-user-pro-upgrade
open "canvas" view
select "MD" screen size
click on hamburger menu button
see that the header does not disappear

screen size selection is done with the squares icon:
Screen Shot 2022-02-08 at 5 51 21 PM

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut-labs/src/AppHeaderMobile/index.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ export type AppHeaderMobileProps = {
 
 const StyledOverlay = styled(Overlay)(
   css({
-    display: { _: `block`, md: `none` },
+    display: { _: `block`, lg: `none` },
```
Comment (saghdaey) [inline packages/gamut-labs/src/AppHeaderMobile/index.tsx:31]: this now matches the widths at line 103

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.61-alpha.988d9e.0<br/>@codecademy/gamut-labs@29.3.2-alpha.988d9e.0<br/>@codecademy/styleguide@54.3.2-alpha.988d9e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6202f17e41228c0b6184f777--gamut-preview.netlify.app>https://6202f17e41228c0b6184f777--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6202f17e41228c0b6184f777>Deploy Logs</a></p>

**Review by jakemhiller — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.61-alpha.68a080.0<br/>@codecademy/gamut-labs@29.3.2-alpha.68a080.0<br/>@codecademy/styleguide@54.3.2-alpha.68a080.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6202f8c9a7e33200bc7a6bc8--gamut-preview.netlify.app>https://6202f8c9a7e33200bc7a6bc8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6202f8c9a7e33200bc7a6bc8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.61-alpha.ddd461.0<br/>@codecademy/gamut-labs@29.3.2-alpha.ddd461.0<br/>@codecademy/styleguide@54.3.2-alpha.ddd461.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62030051d2b82c0b4a6fb5ab--gamut-preview.netlify.app>https://62030051d2b82c0b4a6fb5ab--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62030051d2b82c0b4a6fb5ab>Deploy Logs</a></p>

File: packages/gamut-labs/src/AppHeaderMobile/index.tsx (hunk 1 above)
Comment (saghdaey) [reply packages/gamut-labs/src/AppHeaderMobile/index.tsx:31]: sure! I like that suggestion

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.61-alpha.e862d6.0<br/>@codecademy/gamut-labs@29.3.2-alpha.e862d6.0<br/>@codecademy/styleguide@54.3.2-alpha.e862d6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://620413a9709dae0dfa26817c--gamut-preview.netlify.app>https://620413a9709dae0dfa26817c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/620413a9709dae0dfa26817c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.61-alpha.81b0af.0<br/>@codecademy/gamut-labs@29.3.2-alpha.81b0af.0<br/>@codecademy/styleguide@54.3.2-alpha.81b0af.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://620436764aaa17378cbe14c7--gamut-preview.netlify.app>https://620436764aaa17378cbe14c7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/620436764aaa17378cbe14c7>Deploy Logs</a></p>

**Review by MattSchiller — APPROVED** [review]
:chefkiss:

**Pull request merged**